src/uast.cc

```diff
@@ -248,125 +415,122 @@ static xmlNodePtr CreateXmlNode(const Uast *ctx, void *node,
   int roles_size = 0;
   const char *token = nullptr;
 
-  if (!xmlNode) {
-    goto error;
-  }
-
-  xmlNode->_private = node;
-  if (parent) {
-    if (!xmlAddChild(parent, xmlNode)) {
-      goto error;
+  try {
+    if (!xmlNode) {
+      throw CreateXMLNodeException();
```

Comment:
The error-gotos were causing me more problems to refactor some stuff so I changed it in the new function as since I changed it for the new one I didn't like the inconsistency of using exceptions for error control in one place and gotos in another in the same file I changed both.

Reply:
Nice, I think is a good change, just harder to review both refactoring and new features all at once.